go/vt/topo/shard.go

```diff
@@ -452,12 +450,16 @@ func (si *ShardInfo) UpdateDisableQueryService(ctx context.Context, tabletType t
 		return fmt.Errorf("cannot safely alter DisableQueryService as BlacklistedTables is set")
 	}
 	if !tc.DisableQueryService {
+		// This code is unreachable because we always delete the control record when we enable QueryService.
```

Comment:
I was tempted to. But it's possible some new code could change this without knowing. So, it's good to leave it here as fail-safe.

Reply:
In that case, why don't we panic? If new code gets added then it should just fail bad.

Reply:
We've had some heated arguments in the past about this. The problem with panic is that if tests miss the code path, it will happen in production.

I've now settled on returning an error, and a clarifying comment that the code is unreachable.

The one exception is if the function doesn't return an error. In such cases, I still panic.